app/services/katello/event_queue.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  class EventQueue
+    @event_types = {}
+
+    def self.clear_events(event_type, object_id, on_or_earlier_than)
+      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
+    end
+
+    def self.peek_event
```

Comment:
Works for me
On Jun 15, 2016 12:51 PM, "Eric D Helms" <notifications@github.com> wrote:

> In app/services/katello/event_queue.rb
> <https://github.com/Katello/katello/pull/6123#discussion_r67200511>:
>
> > @@ -0,0 +1,38 @@
> > +module Katello
> > +  class EventQueue
> > +    @event_types = {}
> > +
> > +    def self.clear_events(event_type, object_id, on_or_earlier_than)
> > +      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
> > +    end
> > +
> > +    def self.peek_event
>
> next_event ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Katello/katello/pull/6123/files/83efd9882ca1f8eabd853d6f69f93ed2506eac1a#r67200511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYHRcl7pjseJR_xt_G70xIvfC5-w0fvks5qMC2TgaJpZM4I1UO4>
> .
>